### PR TITLE
Password Reset/Change #29

### PR DESCRIPTION
**Pull request by bryanchriswhite**
### Password reset flow:

User is at bridge-gui login page (click reset password link) —> user is taken to password reset form (type email and new password, and submit) —> user is taken to a confirmation page with a "check your email ..." message.

User gets the email, clicks the link which hits bridge api and then redirects back to bridge-gui login page. The login page shows a "success ... please log in." message (below).

screen shot 2016-06-29 at 11 10 50 am

### CODE REVIEW

**Comment by bryanchriswhite** [comment]
This PR depends on [Bridge PR #127](https://github.com/Storj/bridge/pull/127)

File: src/containers/Auth/LoginForm/index.js [hunk 1]
```diff
@@ -18,42 +18,63 @@ import {reduxForm} from 'redux-form';
 })
 
 export default class LoginForm extends Component {
+  static propTypes = {
+    fields: PropTypes.object.isRequired,
+    error: PropTypes.object,
```
Comment (MrFancyMonocle) [inline src/containers/Auth/LoginForm/index.js]: Getting a warning.js:45Warning: Failed propType: Invalid prop `error` of type `string` supplied to `LoginForm`, expected `object`. Check the render method of `ReduxForm(LoginForm)` here.

File: src/routes.js [hunk 2]
```diff
@@ -2,32 +2,35 @@ import React from 'react';
...
+  App,
+  Auth,
+  EditBucket,
+  NewBucket,
+  Buckets,
+  Dashboard,
+  FileBucket,
+  NotFound,
+  Support
+  // Billing,
```
Comment (MrFancyMonocle) [inline src/routes.js]: Can remove these non-existent routes entirely; feel free to remove any stale references.

File: src/utils/apiClient.js [hunk 3]
```diff
@@ -55,17 +55,15 @@ class WrappedClient extends Client {
...
 
     return requestPromise;
-  };
+  }
 
   resolveFileFromPointers(pointers) {
-    var iteration = 0;
-    var chunks = [];
-    var self = this;
+    const chunks    = [];
```
Comment (MrFancyMonocle) [inline src/utils/apiClient.js]: Let's go ahead and remove all aligned assignment spacing as we go through the project over time; it's proving awkward to maintain as code changes, even if it's pretty to view.

**Comment by MrFancyMonocle** [comment]
Added a few notes inline...

**Comment by bryanchriswhite** [comment]
@MrFancyMonocle nice catches. I've addressed those in 511f309.

I also noticed a hunk in `apiClient` that was accidentally omitted; I squashed it into cecc0d7. This omission was due to the need for several env tweaks to get bridge-gui to play nice with a local bridge server. I'll clean-up these changes and submit a second PR for them. I'll also open an issue for that in the mean time.

**Pull request merged**